### ISSUE TITLE
Are we allowed to use protected for variables in parent classes?

### ISSUE THREAD

**Issue by blmj** [nus-cs2030/2122-s2#73]
Hello,
Just wanted to check if for our PA, we are allowed to set the access modifier to protected for variables in our parent classes so that we can access them from the child class? Or do we have to set all the variables to private and create getters between every level in order to access the properties? Thank you!

**Comment by LauWyeRock** [comment]
Hi! I think prof mentioned that we are to set all our variables to be private final (not too sure) so yea if you want to access properties it'd be better to create getters. I think codecrunch won't allow for other access modifiers too

Also, 'protected' is less restrictive as compared to 'default' and I think prof mentioned we won't be seeing much of 'protected' until we do our project and we have packages to deal with

**Comment by koh-zhe-wei** [comment]
Hello! I believe everything needs to be set to private final according to CodeCrunch so yes getters are the way to go!

**Comment by szeying02** [comment]
Yup, CS2030's CodeCrunch expects `private final` for all variables, so if you really need to access these variables, you'll have to implement getters.

**Comment by jlee1007** [comment]
CS2030 only allows private final (for now?) so do set that for the PA! However, understanding the concept and usage of protected is still important, just not for the lab. Plan properly and create getters for the PA :)

**Comment by Swagston20** [comment]
Everything has to be private final for this mod so have to implement getters
